pkg/controllers/management/etcdbackup/etcdbackup.go

```diff
@@ -236,20 +240,29 @@ func (c *Controller) deleteS3Snapshot(b *v3.EtcdBackup) error {
 		return fmt.Errorf("can't find S3 backup target configuration")
 	}
 	var err error
-	s3Client := &minio.Client{}
+	var s3Client = &minio.Client{}
+	var creds = &credentials.Credentials{}
 	endpoint := b.Spec.BackupConfig.S3BackupConfig.Endpoint
+	bucketLookup := getBucketLookupType(endpoint)
 	bucket := b.Spec.BackupConfig.S3BackupConfig.BucketName
 	// no access credentials, we assume IAM roles
 	if b.Spec.BackupConfig.S3BackupConfig.AccessKey == "" ||
 		b.Spec.BackupConfig.S3BackupConfig.SecretKey == "" {
-		creds := credentials.NewIAM("")
-		s3Client, err = minio.NewWithCredentials(endpoint, creds, true, "")
+		creds = credentials.NewIAM("")
```

Comment:
I want to make sure that Ali cloud does not support we assume the IAM role, right?